technical-reference/reference-technical-articles/technical-doc-token-converter.md

```diff
@@ -0,0 +1,226 @@
+- `tokenAddressOut`: the address of the token the sender wants to receive with the execution of the conversion
+- `to`: the address where the tokens of `tokenAddressOut` should be sent
+
+The received tokens (`amountInMantissa`) are transferred, in the same transaction, to the `destination` address. The `destination` address is configured as a storage attribute in the `AbstractTokenConverter` contract.
+
+If the received amount is less than `amountInMantissa`, or the transferred out amount is less than the expected one, the transaction is reverted, because we don’t support tokens with fees on transferring in this function.
+
+There must be a `ConversionConfig` entry for the pair `tokenAddressIn / tokenAddressOut`. Otherwise, the transaction will be reverted.
+
+**What if there is not enough liquidity?**
+
+- The contract will send to `to` 100% of the available liquidity, and it will only transfer from the sender the needed amount to cover the conversion (that will be less than the `amountInMantissa`)
+- It must satisfied the constraint of sending more than `amountOutMinMantissa` anyway
+- See the second invocation example in the section [Get amount out](https://www.notion.so/TD-26-Token-converters-59622320ee6945e2a8e0e6606af2f8b7?pvs=21)
+
+### Conversions fixing the amount out
+
+Function `convertForExactTokens(uint256 amountInMaxMantissa, uint256 amountOutMantissa, address tokenAddressIn, address tokenAddressOut, address to)`, where:
+
+- `amountInMaxMantissa`: the maximum amount of `tokenAddressIn` tokens that the sender wants to convert. If the needed amount is greater than this param, the transaction is reverted
+- `amountOutMantissa`: the exact amount of `tokenAddressOut` tokens that the sender wants to receive in the conversion
+- `tokenAddressIn`: the address of the token that the sender is providing to complete the conversion
+- `tokenAddressOut`: the address of the token that the sender wants to receive in the conversion
+- `to`: the address where the tokens of `tokenAddressOut` should be sent
+
+The received tokens (`amountInMantissa`) are transferred, in the same transaction, to the `destination` address. The `destination` address is configured as a storage attribute in the `AbstractTokenConverter` contract.
+
+If the received amount or the transferred out amounts are less than the required amounts, the transaction is reverted, because we don’t support tokens with fees on transferring in this function.
+
+There must be a `ConversionConfig` entry for the pair `tokenAddressIn / tokenAddressOut`. Otherwise, the transaction will be reverted.
+
+This function could use internally the `getAmountIn` function. If the `amountConvertedMantissa` amount is less than the param `amountOutMantissa`, the transaction is reverted. This constraint is similar to this one: if the liquidity of `tokenAddressOut` is less than `amountOutMantissa`, then the transaction is reverted.
```

Comment:
```suggestion
This implementation of this function can use the `getAmountIn` function internally because the same liquidity check is needed. If the `amountConvertedMantissa` amount is less than the param `amountOutMantissa`, the transaction is reverted. This constraint is similar to checking that the liquidity of `tokenAddressOut` is less than `amountOutMantissa`.
```